### ISSUE TITLE
s/-/::/ before attempting a module lookup

### ISSUE THREAD

**Issue by oalders** [skaji/cpm#90]
`cpanm` does this and it's very handy, because I can just copy paste a dist name without having to manually do the translation.

eg:

```bash
MaxMind-MacBook:~ olaf$ cpm install -g Mac-PopClip-Quick
FAIL resolve Mac-PopClip-Quick
0 distribution installed.
See /Users/olaf/.perl-cpm/build.log for details.
MaxMind-MacBook:~ olaf$ cpm install -g Mac::PopClip::Quick
DONE install Archive-Zip-1.59
DONE install Mac-PopClip-Quick-1.000001
2 distribution installed.
```

It would have been nice to see that happen on the first attempt.  I can't think of a reason *not* to do this as it shouldn't affect performance and there's no legitimate reason for a `-` to appear in a module name.

I can submit a pull request if you like.

**Comment by skaji** [comment]
Thanks for creating this issue. Let me think of it.

> cpanm does this
```
❯ cpanm Mac-PopClip-Quick
! Finding Mac-PopClip-Quick on cpanmetadb failed.
! Finding Mac-PopClip-Quick () on mirror http://www.cpan.org failed.
! Couldn't find module or a distribution Mac-PopClip-Quick
```
:)

**Comment by oalders** [comment]
@skaji good catch.  ;)  I got the behaviour wrong.  See https://twitter.com/olafalders/status/870012240556064770  Having said that, it would be nice to see both behaviours supported -- ie with and without a trailing `.pm`.  I think it's one of those DWIM things that makes everyone's day a little easier.
